catalog/test/main.js

```diff
@@ -0,0 +1,207 @@
+// Preview user image before upload
+const previewImage = (function() {
+  const makePreview = function(fileInput, img) {
+    const reader = new FileReader();
+
+    reader.onload = e => {
+      img.src =  e.target.result;
+    };
+
+    fileInput.addEventListener('change', e => {
+      const file = e.target.files[0];
+      reader.readAsDataURL(file);
+    });
+  };
+
+  return {
+    makePreview
+  };
+})();
+
+// Get All Plants
+(function getPlants() {
+  const getPlantsBtn = document.getElementById('get-plants-btn');
+
+  const fetchPlants = async () => {
+    const result = await fetch('http://localhost:3000/catalog/plants');
```

Comment:
Avoid hardcoding URLs.

Reply:
This was just for testing purposes. The final implementation with VueJS won't be like that.